src/main/java/core/basesyntax/RightTriangle.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+public class RightTriangle extends Figure {
+    private final double firstLeg;
+    private final double secondLeg;
+    private final ColorSupplier color;
+
+    public RightTriangle(ColorSupplier color, double firstLeg, double secondLeg) {
+        this.color = color;
+        this.firstLeg = firstLeg;
+        this.secondLeg = secondLeg;
+    }
+
+    @Override
+    public double calculateArea() {
+        return (firstLeg * secondLeg) / 2;
+    }
+
+    @Override
+    public void draw() {
+        System.out.println("Figure: rightTriangle, area: "
+                + calculateArea() + "sq. units, firstLeg: "
+                + firstLeg + " units, " + "secondLeg: "
+                + secondLeg + " units, "
+                + "color: " + color);
+    }
+}
```

Comment:
Don't use abstract classes to set behavior for classes. If `Figure` is an abstract class, consider if it should be an interface instead, depending on whether you need to share code (use abstract class) or only enforce a contract (use interface).